snuba_sdk/dsl/dsl.py

```diff
@@ -31,33 +31,34 @@
 coefficient = number / filter
 
 number = ~r"[0-9]+" ("." ~r"[0-9]+")?
-filter = target (open_brace _ condition (_ comma _ condition)* _ close_brace)? (group_by)?
+filter = target (open_brace _ condition (_ comma? _ condition)* _ close_brace)? (group_by)?
 
 condition = condition_op? (variable / tag_key) _ colon _ tag_value
 condition_op = "!"
-tag_key = ~"[a-zA-Z0-9_]+"
-tag_value = quoted_string / quoted_string_tuple / variable
+tag_key = ~r"[a-zA-Z0-9_]+"
+tag_value = quoted_string / unquoted_string / string_tuple / variable
 
-quoted_string = ~r'"([^"\\]*(?:\\.[^"\\]*)*)"'
-quoted_string_tuple = open_square_bracket _ quoted_string (_ comma _ quoted_string)* _ close_square_bracket
+quoted_string = quote unquoted_string quote
+unquoted_string = ~r"[a-zA-Z0-9_]+"
```

Comment:
Related to the comment above, this is too restrictive. This regex will fail if someone tries to match `release_data=2023-01-03T10:00:00` because you aren't allowing dashes or colons. I would recommend, instead of trying to whitelist all the possible allowed characters, maybe blacklist the ones the grammar cares about (commas, colons etc.)

Reply:
Updated with blacklist characters.